connect/runtime/src/test/java/org/apache/kafka/connect/runtime/isolation/VersionedPluginBuilder.java

```diff
@@ -0,0 +1,111 @@
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kafka.connect.runtime.isolation;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+public class VersionedPluginBuilder {
+
+    private static final String VERSION_PLACEHOLDER = "PLACEHOLDER_FOR_VERSION";
+
+    public enum VersionedTestPlugin {
+
+        SINK_CONNECTOR("sampling-connector", "test.plugins.VersionedSamplingSinkConnector"),
+        SOURCE_CONNECTOR("versioned-source-connector", "test.plugins.VersionedSamplingSourceConnector"),
+        CONVERTER("sampling-converter", "test.plugins.VersionedSamplingConverter"),
+        HEADER_CONVERTER("sampling-header-converter", "test.plugins.VersionedSamplingHeaderConverter"),
+        TRANSFORMATION("versioned-transformation", "test.plugins.VersionedTransformation"),
+        PREDICATE("versioned-predicate", "test.plugins.VersionedPredicate");
+
+        private final String resourceDir;
+        private final String className;
+
+        VersionedTestPlugin(String resourceDir, String className) {
+            this.resourceDir = resourceDir;
+            this.className = className;
+        }
+
+        public String resourceDir() {
+            return resourceDir;
+        }
+
+        public String className() {
+            return className;
+        }
+    }
+
+    public static class BuildInfo {
+
+        private final VersionedTestPlugin plugin;
+        private final String version;
+        private String location;
+
+        private BuildInfo(VersionedTestPlugin plugin, String version) {
+            this.plugin = plugin;
+            this.version = version;
+        }
+
+        private void setLocation(String location) {
+            this.location = location;
+        }
+
+        public VersionedTestPlugin plugin() {
+            return plugin;
+        }
+
+        public String version() {
+            return version;
+        }
+
+        public String location() {
+            return location;
+        }
+    }
+
+    private final List<BuildInfo> pluginBuilds;
+
+    public VersionedPluginBuilder() {
+        pluginBuilds = new ArrayList<>();
+    }
+
+    public VersionedPluginBuilder include(VersionedTestPlugin plugin, String version) {
+        pluginBuilds.add(new BuildInfo(plugin, version));
+        return this;
+    }
+
+    public synchronized Path build(String pluginDir) throws IOException {
+        Path pluginDirPath = Files.createTempDirectory(pluginDir);
```

Comment:
This method is leaking files on each run. I found a bunch of files left over on my machine:
```
cd /var/folders/k1/6b93_bm16596yf4d3dlp6vbh0000gn/T
ls
rm -rf SINK_CONNECTOR-* SOURCE_CONNECTOR-* PREDICATE-* TRANSFORMATION-* HEADER_CONVERTER-* CONVERTER-* all_versioned_artifact*
```

Reply:
Thanks for catching this. Added deleteOnExit on the created temp plugins so that it gets cleaned up.